tools/drdeploy.c

```diff
@@ -1676,9 +1676,10 @@ _tmain(int argc, TCHAR *targv[])
     }
 # ifdef UNIX
     /* i#1676: detect whether under gdb */
-    _snprintf(buf, BUFFER_SIZE_ELEMENTS(buf), "/proc/%d/exe", getppid());
-    NULL_TERMINATE_BUFFER(buf);
-    i = readlink(buf, buf, BUFFER_SIZE_ELEMENTS(buf));
```

Comment:
C99 keyword `restrict`: https://en.wikipedia.org/wiki/Restrict

Presumably the compiler objects to the first two arguments to `readlink` being the same pointer.

Reply:
Yep, apparently the The Open Group Base Specifications Issue 7, 2018 edition specifies readlink as having the following prototype: `ssize_t readlink(const char *restrict path, char *restrict buf,
       size_t bufsize)`
(http://pubs.opengroup.org/onlinepubs/9699919799/) and GCC8 warns because of that. On my Linux system readlink is declared without restrict pointers though, so we are probably fine on linux...

Reply:
Interesting.  I guess the workaround of another buffer isn't a big deal in this tool with lots of stack space.